scripts/ci/lib.sh

```diff
@@ -1599,6 +1601,42 @@ post_process_test_results() {
     set -u
 }
 
+gate_flaky_tests() {
+    local exit_code="$1"
+
+    if [[ "${exit_code}" == "0" ]]; then
+        exit "${exit_code}"
+    fi
+
+    # Gating flaky tests is enabled only on PRs.
+    if ! is_in_PR_context; then
+        exit "${exit_code}"
+    fi
+
+    # Prepare flakechecker
+    curl --retry 5 --retry-connrefused -SsfL https://github.com/stackrox/junit2jira/releases/download/v0.0.23/flakechecker -o /tmp/flakechecker && \
+    chmod +x /tmp/flakechecker
```

Comment:
In the unlikely event GitHub is having prolonged trouble
```suggestion
    curl --retry 5 --retry-connrefused -SsfL https://github.com/stackrox/junit2jira/releases/download/v0.0.23/flakechecker -o /tmp/flakechecker || exit "${exit_code}"
    chmod +x /tmp/flakechecker
```